docs/architecture/modernize-desktop/windows-migration.md

```diff
@@ -0,0 +1,524 @@
+Microsoft announced Desktop Bridge (formerly known as Project Centennial).
+Desktop Bridge is a set of tools that allowed developers to bring their existing
+Win32 apps to the Microsoft Store. More capabilities were added in 2017,
+allowing developers to enhance their Win32 apps leveraging some of the new
+Windows 10 APIs, like live tiles and notifications on the action center. But
+still, no new UI controls.
+
+At Build 2018, Microsoft announced a way for developers to use the new Windows
+10 XAML controls into their current Win32 apps, without fully migrating their
+apps to UWP. It was branded as UWP XAML Islands.
+
+### How it works
+
+The Windows 10 1903 update introduces several XAML hosting APIs. Two of them are
+`WindowsXamlManager` and `DesktopWindowXamlSource`.
+
+The `WindowsXamlManager` class handles the UWP XAML Framework. Its
+`InitializeForCurrentThread` method loads the UWP XAML Framework inside the current
+thread of the Win32 app.
+
+The `DesktopWindowXamlSource` is the instance of your XAML Island content. It has
+the `Content` property, which you're responsible for instantiating and setting.
+The `DesktopWindowXamlSource` renders and gets its input from an HWND. It needs to
+know to which other HWND it will attach the XAML Island's one, and you're
+responsible for sizing and positioning the parent's HWND.
+
+WPF or Windows Forms developers don't usually deal with HWND inside their code,
+so it may be hard to understand and handle HWND pointers and the underlying
+wiring stuff to communicate Win32 and UWP worlds.
+
+#### The XAML Islands .NET Wrappers
+
+The Windows Community Toolkit has a set the XAML Islands .NET wrappers for WPF
+or Windows Forms that make easier to use XAML Islands. These wrappers manage the
+HWNDs, the focus management, among other things. Windows Forms and WPF
+developers should use these wrappers.
+
+The Windows Community Toolkit offers two types of controls: Wrapped Controls and
+Hosting Controls
+
+##### Wrapped Controls
+
+These wrapped controls wrap some UWP controls into Windows Forms or WPF
+controls, hiding UWP concepts for those developers. These controls are:
+
+* WebView and WebViewCompatible
+* InkCanvas and InkToolbar
+* MediaPlayerElement
+* MapControl
+
+Add the `Microsoft.Toolkit.Wpf.UI.Controls` package to your project, include the
+reference to the namespace, and start using them.
+
+```xml
+<Window
+        ...
+        xmlns:uwpControls="clr-namespace:Microsoft.Toolkit.Wpf.UI.Controls;assembly=Microsoft.Toolkit.Wpf.UI.Controls">
+<Grid>
+    <Grid.RowDefinitions>
+        <RowDefinition Height="Auto"/>
+        <RowDefinition Height="\*"/>
+    </Grid.RowDefinitions>
+    <uwpControls:InkToolbar TargetInkCanvas="{x:Reference Name=inkCanvas}"/>
+    <uwpControls:InkCanvas Grid.Row="1" x:Name="inkCanvas" />
+</Grid>
+```
+
+##### Hosting controls
+
+The power of XAML Islands extends to most inbox controls, third-party
```

Comment:
There are still some of my previous comments that were marked as outdated but still valid. for example, here:

inbox controls? maybe first-party?